src/pudl/io_managers.py

```diff
@@ -310,27 +308,44 @@ def load_input(self, context: InputContext) -> pd.DataFrame:
 class PudlParquetIOManager(IOManager):
     """IOManager that writes pudl tables to pyarrow parquet files."""
 
-    def handle_output(self, context: OutputContext, obj: pd.DataFrame) -> None:
+    def handle_output(
+        self, context: OutputContext, obj: pd.DataFrame | pl.LazyFrame
+    ) -> None:
         """Writes pudl dataframe to parquet file."""
-        if not isinstance(obj, pd.DataFrame):
-            raise TypeError(f"Only pandas dataframes are supported, got {type(obj)}.")
         table_name = get_table_name_from_context(context)
         parquet_path = PudlPaths().parquet_path(table_name)
         parquet_path.parent.mkdir(parents=True, exist_ok=True)
 
-        res = Resource.from_id(table_name)
-        df = res.enforce_schema(obj)
-        pa_schema = res.to_pyarrow()
-        df.to_parquet(
-            path=parquet_path,
-            index=False,
-            schema=pa_schema,
-        )
+        if isinstance(obj, pd.DataFrame):
+            res = Resource.from_id(table_name)
+            df = res.enforce_schema(obj)
+            pa_schema = res.to_pyarrow()
+            df.to_parquet(
+                path=parquet_path,
+                index=False,
+                schema=pa_schema,
+            )
+        elif isinstance(obj, pl.LazyFrame):
+            logger.warning(
+                "PudlParquetIOManager currently does not do any schema enforcement for polars LazyFrames"
+            )
+            obj.sink_parquet(parquet_path, engine="streaming")
+        else:
+            raise TypeError(
+                "PudlParquetIOManager only supports pandas DataFrames and Polars LazyFrames"
+                f", got {type(obj)}."
+            )
 
-    def load_input(self, context: InputContext) -> pd.DataFrame | gpd.GeoDataFrame:
+    def load_input(
+        self, context: InputContext
+    ) -> pd.DataFrame | gpd.GeoDataFrame | pl.LazyFrame:
         """Loads pudl table from parquet file."""
         table_name = get_table_name_from_context(context)
-        return get_parquet_table(table_name)
+        if context.dagster_type.typing_type == pl.LazyFrame:
+            df = get_parquet_table_polars(table_name)
+        else:
+            df = get_parquet_table(table_name)
+        return df
```

Comment:
The previous version of this was defaulting to returning a `pl.LazyFrame` if the type wasn't explicityly `pd.DataFrame` which created some errors -- we have many assets of type `Any`.  Now it defaults to pandas and only does the LazyFrame if explicitly asked.